version/check.go

```diff
@@ -0,0 +1,69 @@
+package version
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+const versionTimeout = time.Second * 1
+
+// GetLatest returns the latest version if the current version is less than the latest version
+// For Example :
+//    githubReleaseURL = "https://api.github.com/repos/odpf/optimus/releases/latest"
+//    currVersion = 0.0.1
+// the curVersion is the version of the binary installed in user system and the
+// githubReleaseURL will be used to extract the latest version from `json:"tag_name"`
+// suppose, the latestV is 0.0.2 which is greater than currentV, then
+// the GetLatest returns v0.0.2
+func GetLatest(currVersion string, githubReleaseURL string) (string, error) {
+	gitClient := http.Client{
+		Timeout: versionTimeout,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, githubReleaseURL, nil)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create request for latest version")
+	}
+	req.Header.Set("User-Agent", "salt")
```

Comment:
Let's make this a bit more specific. If we do use `repoName` in the function params then we can use that plus the `currVersion`. So something like - 
`odpf/optimus/v0.1.0 (odpf/salt)`